indexing-service/src/main/java/io/druid/indexing/common/task/AppendTask.java

```diff
@@ -46,14 +47,19 @@
  */
 public class AppendTask extends MergeTaskBase
 {
+
+  private final IndexSpec indexSpec;
+
   @JsonCreator
   public AppendTask(
       @JsonProperty("id") String id,
       @JsonProperty("dataSource") String dataSource,
-      @JsonProperty("segments") List<DataSegment> segments
+      @JsonProperty("segments") List<DataSegment> segments,
+      @JsonProperty("indexSpec") IndexSpec indexSpec
   )
   {
     super(id, dataSource, segments);
+    this.indexSpec = indexSpec == null ? new IndexSpec() : indexSpec;
```

Comment:
new IndexSpec() is the default. I didn't feel the need to make it static since it's only used once throughout the class. A new instance will be created anyway if the user provides the configuration.